### PR TITLE
Feat  to do list

### PR DESCRIPTION
**Pull request by eapenzacharias**
Featured and functions added:

- Set up a project with webpack
- Created an index.html file and added HTML markup with empty placeholders
- On page load, the dynamically created list of tasks is rendered into the dedicated placeholder
- Created a style.css from style.scss and set rules for the To-Do List and loaded the CSS using Webpack CSS loader
- Added JavaScript files and imported them as a module for updating completion status.
- Event listener added to the checkbox
- Update item's completed key upon user click
- Store the updated array of items in local storage
- User gets the updated values after the page reloads
- Implemented a function for adding a new task
- Added function for editing task descriptions
- function for deleting a task when clicked on the 'Clear all completed' button
- function for deleting individual tasks
- Updates indexes if a task is deleted
- all changes are updated in localStorage

### CODE REVIEW

File: src/scripts/updateIndex.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { updateLocal } from './localStorage.js';
+
+function newIndex(tasks) {
+  for (let i = 0; i < tasks.length; i += 1) {
```
Comment (algerina) [inline src/scripts/updateIndex.js]: It's better to use the forEach method instead of For loop.

File: src/scripts/updateTasks.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  if (done.checked) {
+    tasks[objIndex].completed = true;
+    updateLocal(tasks);
+  } else {
+    tasks[objIndex].completed = false;
+    updateLocal(tasks);
+  }
+}
+
+function editTask(editBtn, description, tasks, task) {
```
Comment (algerina) [inline src/scripts/updateTasks.js]: - [x] [OPTIONAL]   It's better to void a long number of arguments. Ideally, a function should have two or fewer arguments specified. The fewer the arguments, the easier is to test the function.

**Review by algerina — APPROVED** [review]
Hi, 
Good job so far! @eapenzacharias 

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your project looks awesome:heavy_check_mark:
-  Your code is Dry :heavy_check_mark:




Great work, however, I have some suggestions.
 
 ## Optional suggestions
 
 - Please check the comment under the review.


Cheers and Happy coding! :clap: :clap: :clap:

**Comment by eapenzacharias** [comment]
Thank you for your feedback.

**Pull request merged**